util/exec_test.go

```diff
@@ -91,3 +91,39 @@ func TestFilterArgs(t *testing.T) {
 		},
 	})
 }
+
+func TestParallelExecute(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Timeout time.Duration
+		Limit   uint
+		Times   uint
+
+		Function func()
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+
+			p := NewParallel(tc.Limit, tc.Timeout)
+
+			for i := uint(0); i <= tc.Times; i++ {
+				p.Execute(tc.Function)
+			}
+			p.Wait()
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Run 10 times with limit of 2",
+
+		Timeout: 0,
+		Limit:   2,
+		Times:   10,
+
+		Function: func() {
+			// do something?
+		},
+	})
```

Comment:
Would have one test case where every function adds `+1` to some mutex and then you check if really all 10x of them ran.
And a second case maybe where you have `time.Sleep(1)` in every function with `Limit:1` and `Times:3` and make sure the whole thing runs for at least like `2.5` seconds.

Reply:
The first one sounds right and enough, I do not want to use timed counts with `sleep` as they artificially increase execution time where it is not needed, as in fact both do the same thing only the measurement is different.

Reply:
mhmmm... if you don't want sleeps than I am not sure how to easily measure that only x run at a time.

Reply:
Hmm yeah you are right, testing the limit is only possible with time. Fixed that up. Please review again @bauersimon